Clarify terminology plugin/device in docs (#770)
PennyLaneAI/pennylane: .github/CHANGELOG.md

  ```

  See the `default.qubit.tf`

>   [documentation](https://pennylane.ai/en/stable/code/api/pennylane.beta.devices.DefaultQubitTF.html)

Author: ```suggestion
  [documentation](https://pennylane.ai/en/stable/code/api/pennylane.beta.plugins.DefaultQubitTF.html)
```

  [documentation](https://pennylane.ai/en/stable/code/api/pennylane.beta.plugins.DefaultQubitTF.html)
  for more details.

> * The [default.tensor device](https://github.com/XanaduAI/pennylane/blob/master/pennylane/beta/devices/default_tensor.py)

Author: ```suggestion
* The [default.tensor device](https://github.com/XanaduAI/pennylane/blob/master/pennylane/beta/plugins/default_tensor.py)
```

> * The [default.tensor device](https://github.com/XanaduAI/pennylane/blob/master/pennylane/beta/plugins/default_tensor.py)

Author: ```suggestion
* The [default.tensor device](https://github.com/XanaduAI/pennylane/blob/master/pennylane/beta/devices/default_tensor.py)
```

> * The [default.tensor device](https://github.com/XanaduAI/pennylane/blob/master/pennylane/beta/plugins/default_tensor.py)

Author: ```suggestion
* The [default.tensor plugin](https://github.com/XanaduAI/pennylane/blob/master/pennylane/beta/plugins/default_tensor.py)
```

  of an operator in the computational basis.
  [(#454)](https://github.com/XanaduAI/pennylane/pull/454)

> * Added a `QubitDevice` class which implements common functionalities of devices such that
>   plugins can rely on these implementations. The new `QubitDevice` also includes
>   a new `execute` method, which allows for more convenient device design. In addition, `QubitDevice`

Author: ```suggestion
* Added a `QubitDevice` class which implements common functionalities of plugin devices such that
  plugin devices can rely on these implementations. The new `QubitDevice` also includes
  a new `execute` method, which allows for more convenient plugin design. In addition, `QubitDevice`
```

    which will alert the user that an operation with `num_wires = All`
    is being incorrectly.

> * The one-qubit rotations in `pennylane.devices.default_qubit` no longer depend on Scipy's `expm`. Instead

Author: ```suggestion
* The one-qubit rotations in `pennylane.plugins.default_qubit` no longer depend on Scipy's `expm`. Instead
```

* Removed use of hardcoded values in the optimizers, made them parameters (see [#131](https://github.com/XanaduAI/pennylane/pull/131) and [#132](https://github.com/XanaduAI/pennylane/pull/132))
* Created the new `PlaceholderExpectation`, to be used when both CV and qubit expval modules contain expectations with the same name

> * Provide a way for plugins to view the operation queue _before_ applying operations. This allows for on-the-fly modifications of

Author: Just fixing this error here
